### PR TITLE
Fix 7671 - shift+click should add a disabled breakpoint

### LINKED ISSUE #7671: [breakpoints] shift+click should add a disabled breakpoint
When the user shift+clicks in the gutter, the debugger should create a disabled breakpoint. Currently it creates an enabled breakpoint.

This is a good starting off spot

```diff
diff --git a/src/actions/breakpoints/index.js b/src/actions/breakpoints/index.js
index 830b39a64..4aa386967 100644
--- a/src/actions/breakpoints/index.js
+++ b/src/actions/breakpoints/index.js
@@ -385,7 +385,11 @@ export function toggleBreakpointsAtLine(line: number, column?: number) {
   };
 }

-export function addOrToggleDisabledBreakpoint(line: number, column?: number) {
+export function addOrToggleDisabledBreakpoint(
+  line: number,
+  column?: number,
+  shouldDisable
+) {
   return ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
     const selectedSource = getSelectedSource(getState());

@@ -411,7 +415,8 @@ export function addOrToggleDisabledBreakpoint(line: number, column?: number) {
         sourceId: selectedSource.id,
         sourceUrl: selectedSource.url,
         line: line,
-        column: column
+        column: column,
+        enabled: !shouldDisable
       })
     );
   };
diff --git a/src/components/Editor/index.js b/src/components/Editor/index.js
index 7227c11d3..e0cf932a4 100644
--- a/src/components/Editor/index.js
+++ b/src/components/Editor/index.js
@@ -404,7 +404,7 @@ class Editor extends PureComponent<Props, State> {
     }

     if (ev.shiftKey) {
-      return addOrToggleDisabledBreakpoint(sourceLine);
+      return addOrToggleDisabledBreakpoint(sourceLine, ev.shiftKey);
     }
:
```

### PR DESCRIPTION
**Pull request by darkwing**
Fixes #7671 

Supercedes: https://github.com/devtools-html/debugger.html/pull/7694

### CODE REVIEW

**Comment by darkwing** [comment]
@baronKilljoy Let's iterate on this.  We should add a test for sure.  I'd love you to manually test this too :)

**Comment by baronKilljoy** [comment]
@darkwing is there another fix mixed here, the base code is different than mine. What is the !! in !!ev.shiftKey. If shiftkey isn't pressed ev.shiftKey is undefined not false.

**Comment by baronKilljoy** [comment]
@darkwing without my other changed to addBreakpointPromise the checkbox on the secondary box breaks. here is an alternate fix to get rid of the need for the redundant code. 
altfix
The trouble is `export function addBreakpoint(shiftKeyPress,
  location: SourceLocation,
  { condition, hidden }: addBreakpointOptions = {}
) `
addBreakpoint options having a default value means we need to add the bool shiftKeypress cant be defaulted. addBreakpoint ill need to be changed everywhere its called

**Comment by darkwing** [comment]
This is the weirdness that tests caught: oO

wtfbreakpoints

When I click the checkbox, not only does the breakpoint not get enabled, it shifts locations...

File: src/actions/breakpoints/addBreakpoint.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ async function addBreakpointPromise(getState, client, sourceMaps, breakpoint) {
 
   const newBreakpoint = {
     id: makeLocationId(generatedLocation),
-    disabled: false,
+    disabled: breakpoint.disabled,
```
Comment (darkwing) [inline src/actions/breakpoints/addBreakpoint.js]: `breakpoint` is the data with options, which is why I'm using its `disabled` property.  We can't simply default this to `false`

**Comment by darkwing** [comment]
@jasonLaster Can you think of why initializing a breakpoint to `disabled: true` would cause location shifting and likewise weirdness?

**Comment by darkwing** [comment]
@bomsy @AnshulMalik @princiya @jainsneha23  I'd love your eyes on this as well.

File: src/actions/breakpoints/addBreakpoint.js (hunk 1 above)
Comment (baronKilljoy) [reply src/actions/breakpoints/addBreakpoint.js]: newBreakpoint needs to be false or the checkbox breaks.  thats why i manually set it to false if the shiftkey is pressed in my code.

**Comment by jasonLaster** [comment]
@darkwing nope

**Comment by darkwing** [comment]
I did some very advanced console.logging here (/me hides) and found something really weird:

what in the

When clicking on the checkbox to enable the breakpoint, the breakpoint's `id` changes, removing `originalSource` and then losing the line number at the end.

File: src/actions/breakpoints/addBreakpoint.js (hunk 1 above)
Comment (darkwing) [reply src/actions/breakpoints/addBreakpoint.js]: Right; this is the current problem; it's as though this is resetting the value somehow

**Comment by darkwing** [comment]
This is a step further in the right direction.  There's only one case that's now causing weirdness:

1.  Shift-click to create a disabled breakpoint
2.  Click its checkbox in the breakpoints pane

Expected:

Breakpoint is enabled

Actual:

The breakpoint is disappearing completely, and `dbg.selectors.getBreakpointsList()` returns an empty array.  oO

**Comment by baronKilljoy** [comment]
@jaril  said some thing that may be relevant in the original bug.
> My understanding is the client doesn't retain whether a breakpoint is disabled or not. It only either 1) sets breakpoints or 2) removes breakpoints to pause on. This is in `/src/client/firefox/commands.js`.
> 
> When a breakpoint (that's registered in the redux store and displayed in the Breakpoints pane) is skipped, it's because that breakpoint has been removed from the separate list of breakpoints set for the client to look at and pause on. What looks like toggling enable/disable breakpoints in the UI actually adds/removes breakpoints on the client.

>

**Pull request closed without merging**